src/Eris/Generator/SuchThatGenerator.php

```diff
@@ -3,21 +3,25 @@
 
 use Eris\Generator;
 use LogicException;
+use PHPUnit_Framework_Constraint;
+use PHPUnit_Framework_ExpectationFailedException;
 
 /**
+ * @param callable|PHPUnit_Framework_Constraint $filter
  * @return SuchThatGenerator
  */
-function filter(callable $filter, Generator $generator)
+function filter($filter, Generator $generator, $maximumAttempts = 100)
 {
```

Comment:
@giorgiosironi 

This will probably need a pass of

```
$ composer cs
``` 

as the opening brace for a function should be placed on the same line, e.g.

```php
function filter($filter, Generator $generator, $maximumAttempts = 100) {
    // ...
}
```